src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -0,0 +1,945 @@
+                        }
+
+                        break;
+
+                    case "CharSet":
+                        // invalid values will be ignored
+                        charSet = namedArg.Value.DecodeValue<CharSet>(SpecialType.System_Enum);
+                        break;
+
+                    case "SetLastError":
+                        // invalid values will be ignored
+                        setLastError = namedArg.Value.DecodeValue<bool>(SpecialType.System_Boolean);
+                        break;
+
+                    case "ExactSpelling":
+                        // invalid values will be ignored
+                        exactSpelling = namedArg.Value.DecodeValue<bool>(SpecialType.System_Boolean);
+                        break;
+
+                    case "PreserveSig":
+                        preserveSig = namedArg.Value.DecodeValue<bool>(SpecialType.System_Boolean);
+                        break;
+
+                    case "CallingConvention":
+                        // invalid values will be ignored
+                        callingConvention = namedArg.Value.DecodeValue<CallingConvention>(SpecialType.System_Enum);
+                        break;
+
+                    case "BestFitMapping":
+                        bestFitMapping = namedArg.Value.DecodeValue<bool>(SpecialType.System_Boolean);
+                        break;
+
+                    case "ThrowOnUnmappableChar":
+                        throwOnUnmappable = namedArg.Value.DecodeValue<bool>(SpecialType.System_Boolean);
+                        break;
+                }
+
+                position++;
+            }
+
+            if (!hasErrors)
+            {
+                arguments.GetOrCreateData<MethodWellKnownAttributeData>().SetDllImport(
+                    arguments.Index,
+                    moduleName,
+                    importName,
+                    DllImportData.MakeFlags(
+                        exactSpelling,
+                        charSet,
+                        setLastError,
+                        callingConvention,
+                        bestFitMapping,
+                        throwOnUnmappable),
+                    preserveSig);
+            }
+        }
+
+        internal sealed override void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttributeData> boundAttributes, ImmutableArray<AttributeSyntax> allAttributeSyntaxNodes, DiagnosticBag diagnostics, AttributeLocation symbolPart, WellKnownAttributeData decodedData)
+        {
+            Debug.Assert(!boundAttributes.IsDefault);
+            Debug.Assert(!allAttributeSyntaxNodes.IsDefault);
+            Debug.Assert(boundAttributes.Length == allAttributeSyntaxNodes.Length);
+            Debug.Assert(symbolPart == AttributeLocation.None || symbolPart == AttributeLocation.Return);
+
+            if (symbolPart != AttributeLocation.Return)
+            {
+                Debug.Assert(_lazyCustomAttributesBag != null);
```

Comment:
>Debug.Assert(_lazyCustomAttributesBag != null); [](start = 16, length = 47)

It feels like this is the right place to report the WRN_ExternMethodNoImplementation diagnostics.
```
/// <summary>
/// Called to report attribute related diagnostics after all attributes have been bound and decoded.
/// Called even if there are no attributes.
/// </summary>
```